pitmutationmate/src/main/kotlin/com/amos/pitmutationmate/pitmutationmate/actions/ContextMenuAction.kt

```diff
@@ -5,33 +5,74 @@ package com.amos.pitmutationmate.pitmutationmate.actions
 
 import com.intellij.openapi.actionSystem.AnActionEvent
 import com.intellij.openapi.actionSystem.CommonDataKeys
-import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.PsiClass
 import com.intellij.psi.PsiClassOwner
+import com.intellij.psi.PsiElement
+import org.jetbrains.kotlin.psi.KtClass
 
 class ContextMenuAction : RunConfigurationAction() {
```

Comment:
The class give the warning `Override 'getActionUpdateThread' and choose 'EDT' or 'BGT'`

The docs state that `getActionUpdateThread` should be overridden and a ActionUpdateThread should be chosen, which should be the BGT since this does not access any Swing stuff right?
https://plugins.jetbrains.com/docs/intellij/basic-action-system.html#principal-implementation-overrides

> AnAction.getActionUpdateThread() (2022.3+) return an [ActionUpdateThread](https://github.com/JetBrains/intellij-community/tree/idea/233.11799.241/platform/editor-ui-api/src/com/intellij/openapi/actionSystem/ActionUpdateThread.java), which specifies if the update() method is called on a [background thread (BGT) or the event-dispatching thread (EDT)](https://plugins.jetbrains.com/docs/intellij/general-threading-rules.html). The preferred method is to run the update on the BGT, which has the advantage of guaranteeing application-wide read access to [PSI](https://plugins.jetbrains.com/docs/intellij/psi.html), [the virtual file system](https://plugins.jetbrains.com/docs/intellij/virtual-file-system.html) (VFS), or [project models](https://plugins.jetbrains.com/docs/intellij/project-structure.html). Actions that run the update session on the BGT should not access the Swing component hierarchy directly. Conversely, actions that specify to run their update on the EDT must not access PSI, VFS, or project data but have access to Swing components and other UI models. All accessible data is provided by the DataContext as explained in [Determining the Action Context](https://plugins.jetbrains.com/docs/intellij/basic-action-system.html#determining-the-action-context). When switching from BGT to EDT is absolutely necessary, actions can use AnActionEvent.getUpdateSession() to access the [UpdateSession](https://github.com/JetBrains/intellij-community/tree/idea/233.11799.241/platform/editor-ui-api/src/com/intellij/openapi/actionSystem/UpdateSession.java) and then call UpdateSession.compute() to run a function on the EDT. Starting from IntelliJ Platform version 2022.3, the Plugin DevKit will have an inspection in Plugin DevKit | Code | ActionUpdateThread is missing to notify plugin authors about a missing implementation of AnAction.getActionUpdateThread().